### PR TITLE
Update space module docstring

### PR DESCRIPTION
**Pull request by Tortar**
I added the missing classes and modified (hopefully improving) the wording about the ones already present.

### CODE REVIEW

**Comment by rht** [comment]
LGTM

**Comment by rht** [comment]
I only saw the first commit just now. I have some comments.

File: mesa/space.py [hunk 1]
```diff
@@ -4,9 +4,12 @@
 
 Objects used to add a spatial component to a model.
 
-Grid: base grid, a simple list-of-lists.
-SingleGrid: grid which strictly enforces one object per cell.
-MultiGrid: extension to Grid where each cell is a set of objects.
+Grid: base grid, it creates a square grid.
```
Comment (rht) [inline mesa/space.py:7]: Should be rectangular. I know the `Grid` docstring says square, which should be fixed as well.

Comment (Tortar) [reply mesa/space.py:7]: right, I blindly copy-pasted x)

Comment (Tortar) [reply mesa/space.py]: done, fixed also the docstring in the Grid class

File: mesa/space.py [hunk 2]
```diff
@@ -4,9 +4,12 @@
...
+SingleGrid: extension to Grid which strictly enforces one agent per cell.
+MultiGrid: extension to Grid where each cell can contain a set of agents.
+HexGrid: extension to Grid to handle hexagonal neighbors.
+ContinuousSpace: a nondiscrete space where each agent can have an arbitrary position.
```
Comment (rht) [inline mesa/space.py:11]: I think it is more precise to mention float position somehow here. But this version is fine too for now.

Comment (Tortar) [reply mesa/space.py:11]: "arbitrary float position"?

Comment (rht) [reply mesa/space.py:11]: I was about to say that, but the position is a tuple `tuple[float, float]`.

Comment (Tortar) [reply mesa/space.py:11]: mmh "two dimensional arbitrary float position"?

Comment (rht) [reply mesa/space.py:11]: What about "a two-dimensional space where each agent has an arbitrary position of `float`'s"?

Comment (Tortar) [reply mesa/space.py]: sounds good, made it that way

File: mesa/space.py [hunk 3]
```diff
@@ -4,9 +4,13 @@
...
 
-Grid: base grid, a simple list-of-lists.
-SingleGrid: grid which strictly enforces one object per cell.
-MultiGrid: extension to Grid where each cell is a set of objects.
+Grid: base grid, it creates a rectangular grid.
+SingleGrid: extension to Grid which strictly enforces one agent per cell.
+MultiGrid: extension to Grid where each cell can contain a set of agents.
+HexGrid: extension to Grid to handle hexagonal neighbors.
+ContinuousSpace: a two-dimensional space where each agent has an arbitrary
+position of `float`'s.
```
Comment (rht) [inline mesa/space.py:12]: We can improve the wording more later. Though this line needs to be indented
```
ContinuousSpace: a two-dimensional space where each agent has an arbitrary
                              position of `float`'s.
```

Comment (Tortar) [reply mesa/space.py]: done

File: mesa/space.py [hunk 4]
```diff
@@ -4,9 +4,13 @@
 
 Objects used to add a spatial component to a model.
 
-Grid: base grid, a simple list-of-lists.
-SingleGrid: grid which strictly enforces one object per cell.
-MultiGrid: extension to Grid where each cell is a set of objects.
+Grid: base grid, it creates a rectangular grid.
```
Comment (rht) [inline mesa/space.py:7]: Should be `base grid, which creates a rectangular grid`.

Comment (Tortar) [reply mesa/space.py]: 👍

**Comment by rht** [comment]
LGTM

**Comment by Tortar** [comment]
I fixup then

**Comment by rht** [comment]
I tweaked the commit message to "Update space module-level docstring summary".

**Pull request merged**